apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -2026,6 +2094,66 @@ handle_get_data_sync_record(Start, End, Limit, Req) ->
 			{503, #{}, jiffy:encode(#{ error => timeout }), Req}
 	end.
 
+handle_get_footprints(Partition, FootprintNumber, Req) ->
+	{Start, End} = ar_replica_2_9:get_entropy_partition_range(Partition),
+	FindStorageModules =
+		case ar_storage_module:get_all(Start, End) of
+			[] ->
+				{404, #{}, <<>>, Req};
+			Modules ->
+				{ok, Modules}
+		end,
+	FindStoreIDPacking =
+		case FindStorageModules of
+			{ok, StorageModules} ->
+				{_, _, Packing} = StorageModule = hd(StorageModules),
+				StoreID = ar_storage_module:id(StorageModule),
+				{ok, {StoreID, Packing}};
+			Reply ->
+				Reply
+		end,
+	CheckFootprintNumber =
+		case FindStoreIDPacking of
+			{ok, {StoreID2, Packing2}} ->
+				FootprintSize = ?REPLICA_2_9_ENTROPY_SIZE div ?COMPOSITE_PACKING_SUB_CHUNK_SIZE,
+				FootprintsPerPartition = ?REPLICA_2_9_ENTROPY_COUNT div ?COMPOSITE_PACKING_SUB_CHUNK_COUNT,
+				case FootprintNumber >= FootprintsPerPartition of
+					true ->
+						{400, #{}, jiffy:encode(#{ error => footprint_number_too_large }), Req};
+					false ->
+						{ok, {StoreID2, Packing2, FootprintSize, FootprintsPerPartition}}
+				end;
+			Reply2 ->
+				Reply2
+		end,
+	case CheckFootprintNumber of
+		{ok, {StoreID3, Packing3, FootprintSize2, FootprintsPerPartition2}} ->
+			PartitionStartOffset = Partition * FootprintsPerPartition2 * FootprintSize2,
```

Comment:
Tagging in case similar comment applies: https://github.com/ArweaveTeam/arweave/pull/812/files#r2103123546